src/app/Dashboard/Topic.js

```diff
@@ -33,24 +43,52 @@ export class Topic extends React.Component {
     const { title, topicId } = this.props;
 
     return (
-      <div
-        className="topic-wrap"
-        onMouseEnter={this.toggleHidden}
-        onMouseLeave={this.toggleHidden}
-      >
-        <button
-          className="topic-btn"
-          onClick={() => console.log('click through to topic/:id', topicId)}
+      <>
+        <div
+          className="topic-wrap"
+          onMouseEnter={this.toggleHidden}
+          onMouseLeave={this.toggleHidden}
         >
-          {title}
-        </button>
-        {!this.state.isHidden && (
-          <div className="edit-delete-topic-options">
-            <button onClick={this.editTopic}>Edit</button>
-            <button onClick={this.deleteTopic}>Delete</button>
-          </div>
-        )}
-      </div>
+          {this.state.editing ? (
+            <>
+              <form
+                className="edit-topic-form"
+                onSubmit={e => this.submitEdit(e)}
```

Comment:
If you make submitEdit an arrow function (to prevent the `this` context from being ambiguous), you can change this line to be like your other event handlers:

```js
-  onSubmit={e => this.submitEdit(e)}
+  onSubmit={this.submitEdit}
```